### ISSUE TITLE
testing on pkg.julialang.org fails

### ISSUE THREAD

**Issue by lobingera** [JuliaPlots/Plots.jl#243]
http://pkg.julialang.org/detail/Plots.html

shows failing Pkg.test. afaics, a missing dependency

> ERROR: LoadError: LoadError: ArgumentError: VisualRegressionTests not found in path

**Comment by tbreloff** [comment]
This is a really old version of Plots.  I certainly need to tag a new version... any idea how to update this page once I do?

**Comment by lobingera** [comment]
The page will be updated with https://github.com/JuliaCI/PackageEvaluator.jl runs an afaiu this is done manually (and recently rather seldom OR the webpage update was stopped). You can check their FAQ about other things that could fail the tests. I just mention this here, as some people will look at Plots.jl in pkg.julialang.org and then maybe decide not to use it, when test fail...

**Comment by tbreloff** [comment]
I agree that I'd prefer this to be passing, but it seems like a tricky target to hit when the pulse is only run once every couple of months on an old Plots version.

Part of the trickiness is that I have extensive tests driven by VisualRegressionTests, and they need to be in sync with ExamplePlots.jl, where all the reference images are currently being stored.  I'm not sure of a good solution to that (and I don't want to remove tests, just so they pass)

**Comment by lobingera** [comment]
(@iainNZ) The description in  https://github.com/JuliaCI/PackageEvaluator.jl starts with
> The purpose of PackageEvaluator is to attempt to test every Julia package nightly, 
so i don't think it's the plan to run this only once a few weeks...

**Comment by IainNZ** [comment]
Its meant to be run more frequently, but I'm the only one really working on it and I've been too busy to figure out phantom issues on 0.5.

However, the fact that didn't work when it ran early this week doesn't really suggest a problem that'd be resolved by more frequent runs. If a user can't run Pkg.test on their own machine without having a bunch of stuff checked out to master, then PkgEval isn't going to work. If you don't like how it looks, you can opt out at any time.

**Comment by tbreloff** [comment]
>  If a user can't run Pkg.test on their own machine without having a bunch of stuff checked out to master

This is certainly tricky considering the large range of software that has to be working well (and in sync), and the fact that the reference images are stored in a separate repo which tracks master.  I think it's likely that tests fail even if everything would work well for a user.

I think it's fine to keep it as-is, though if it's an option to turn the tests off but keep Plots in the package list, that would be best.  Thanks.

**Comment by IainNZ** [comment]
When you tag a new version of Plots.jl, tag a new version of the reference images, and make sure that the `test/REQUIRE` files specifies exactly that version - seems like it'd work fine.

Instructions for how to optout are in the README. Or just PR a line in https://github.com/JuliaCI/PackageEvaluator.jl/blob/master/src/constants.jl for any of your packages you want to exclude.

**Comment by diegozea** [comment]
Could it work adding something like the following to `runtests.jl`?
```julia
try
using ExamplePlots

**Comment by tbreloff** [comment]
See: https://github.com/JuliaCI/PackageEvaluator.jl/pull/125

Your suggestion of tagging and tracking versions in the `test/REQUIRE` file is a good one, but will require some thought to make sure I do it right.  If I figure out how to get the tests to pass consistently (without resorting to ignoring the tests) then I'll submit a new PR to get them running again.

Thanks @IainNZ

**Comment by tkelman** [comment]
The correct way to modify package states for running tests is `test/REQUIRE`. Do not do any other Pkg operations in `deps/build.jl` or tests.

**Comment by lobingera** [comment]
```
Julia v0.4

2016-06-10 to 2016-06-14, v0.7.0, Package was untestable.
```

doesn't look promising...

**Comment by tkelman** [comment]
that's because of https://github.com/JuliaCI/PackageEvaluator.jl/pull/125

**Comment by tbreloff** [comment]
Though I might switch that back on. I made changes to VisualRegressionTests
to be able to save to a different location than the existing reference, and
I moved reference images for plots to a new repo with versioned directories
and added logic to compared to the most recent image for that version.

All that to say that the tests might pass, IF I remember to bump my version
constant after each tagged release.

Some of this stuff is Plots-only... Eventually I'll move it into
VisualRegressionTests so we can use it for other graphics packages.
